chaco/plots/scatterplot_1d.py

```diff
@@ -32,7 +32,9 @@ class ScatterPlot1D(Base1DPlot):
     marker = MarkerTrait
 
     # The pixel size of the marker, not including the thickness of the outline.
-    marker_size = Float(4.0)
+    marker_size = Union(Float, Int,
+                        Array(Float), Array(Int),
```

Comment:
This isn't how the `Array` trait works - you need to supply a dtype as the argument, not a trait type. As it stands, this would accept an array with any dtype.

Reply:
Got it, thanks!